hypervisor/dm/vpci/vdev.c

```diff
@@ -137,7 +145,11 @@ static void pci_vdev_update_vbar_base(struct pci_vdev *vdev, uint32_t idx)
 		}
 	}
 
-	/* TODO: 1. check whether the address locate in the MMIO windows 2. base must aligned with size */
+	if ( (base != 0UL) && (!is_pci_io_bar(vbar))) {
+		if ((!is_pci_mem_bar_base_valid(vpci2vm(vdev->vpci), base)) || (!mem_aligned_check(base, vdev->vbars[idx].size))) {
+			base = 0UL;
```

Comment:
Hi yuhong

Would you please (1) add error message here; (2) remove the base = 0UL;